R/tab_create_modify.R

```diff
@@ -236,65 +244,43 @@ tab_spanner <- function(data,
 resolve_spanner_level <- function(
   data,
   column_names,
-  spanner_ids,
   level
 ) {
 
-  any_existing_spanners <- dt_spanners_exists(data = data)
-
-  # If explicitly providing a `level`, ensure it's at least 1 or above
+  # If explicitly providing a `level` simply return that value
   if (!is.null(level)) {
-
-    if (level < 1) {
-      stop(
-        "A spanner level of ", level, " cannot be set:\n",
-        "* Please choose a `level` value greater than or equal to `1`",
-        call. = FALSE
-      )
-    }
-
     return(as.integer(level))
   }
 
-  # If spanning exclusively over column labels, and there aren't any
-  # existing spanners, then the new spanner level is always `1`
-  if (length(spanner_ids) < 1 && !any_existing_spanners) {
+  # Determine if there are any existing spanners
+  any_existing_spanners <- dt_spanners_exists(data = data)
+
+  # If there aren't any existing spanners, then the new spanner
+  # level will always be `1`
+  if (!any_existing_spanners) {
     return(1L)
   }
 
-  spanners_existing <- dt_spanners_get(data = data)
+  # Get the present `spanners_tbl`
+  spanners_tbl <- dt_spanners_get(data = data)
 
-  if (length(spanner_ids) < 1 && any_existing_spanners) {
+  highest_level <- 0L
 
-    highest_level <- 0L
+  spanners_tbl <- dplyr::select(spanners_tbl, spanner_id, vars, spanner_level)
 
-    spanners_existing <-
-      spanners_existing %>%
-      dplyr::select(spanner_id, vars, spanner_level)
+  for (i in seq_len(nrow(spanners_tbl))) {
```

Comment:
Just a thought:
```
highest_level <- spanners_tbl %>%
  dplyr::filter(vapply(vars, function(x) any(spanned_vars %in% x), logical(1))) %>%
  dplyr::pull("spanner_level") %>%
  max()
```

Reply:
I was able to get this to work with the current tests in place. Could you make this a code suggestion? Perhaps I didn't integrate this correctly.